src/eth/TransactionManager.js

```diff
@@ -12,12 +15,69 @@ export default class TransactionManager extends PublicService {
   }
 
   formatHybridTx(contract, key, args, name, businessObject = null) {
+
+    // Strip the key to just the method name (in cases where the method was called
+    // using the full method sig e.g. contract["draw(address,uint256)"](foo, bar)
+    const method = key.replace(/\(.*\)$/g, '');
+    const metadata = { contract: name, method, args };
+    let dsProxyAddress = null;
+
+    if (typeof args !== 'undefined' && Array.isArray(args) && typeof args[args.length - 1] === 'object') {
```

Comment:
can you pop the args-manipulating into its own function and add some tests for just that piece? ideally someone would be able to read the titles of those tests to get a quick idea of what the function accomplishes